src/main/scala/org/broadinstitute/monster/etl/encode/EncodeExtractions.scala

```diff
@@ -0,0 +1,216 @@
+package org.broadinstitute.monster.etl.encode
+
+import java.io.{IOException, InputStream, OutputStream}
+import java.nio.ByteBuffer
+
+import com.google.common.io.ByteStreams
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.transforms._
+import com.spotify.scio.values.SCollection
+import com.squareup.okhttp.{Callback, OkHttpClient, Request, Response}
+import io.circe.JsonObject
+import io.circe.jawn.JawnParser
+import io.circe.syntax._
+import org.apache.beam.sdk.coders.{Coder => BeamCoder}
+import org.apache.beam.sdk.coders.{KvCoder, StringUtf8Coder}
+import org.apache.beam.sdk.transforms.{GroupIntoBatches, ParDo}
+import org.apache.beam.sdk.util.VarInt
+import org.apache.beam.sdk.values.KV
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+
+/** Ingest step responsible for pulling raw metadata for a specific entity type from the ENCODE API. */
+object EncodeExtractions {
+
+  // TODO when updating to upack and msg, will need to change this coder as well
+  implicit val jsonCoder: Coder[JsonObject] = Coder.beam {
+    new BeamCoder[JsonObject] {
+      val parser = new JawnParser()
+
+      override def encode(value: JsonObject, outStream: OutputStream): Unit =
+        Option(value).foreach { jsonObj =>
+          val bytes = jsonObj.asJson.noSpaces.getBytes
+          VarInt.encode(bytes.length, outStream)
+          outStream.write(bytes)
+        }
+      override def decode(inStream: InputStream): JsonObject = {
+        val numBytes = VarInt.decodeInt(inStream)
+        val bytes = new Array[Byte](numBytes)
+        ByteStreams.readFully(inStream, bytes)
+        parser.decodeByteBuffer[JsonObject](ByteBuffer.wrap(bytes)).right.get
+      }
+      override def getCoderArguments: java.util.List[_ <: BeamCoder[_]] =
+        java.util.Collections.emptyList()
+      override def verifyDeterministic(): Unit =
+        throw new BeamCoder.NonDeterministicException(
+          this,
+          "Equal Msgs don't necessarily encode to the same bytes"
+        )
+    }
+  }
+
+  /** HTTP client to use for querying ENCODE APIs. */
+  val client = new OkHttpClient()
+
+  /**
+    * Pipeline stage which maps batches of query params to output payloads
+    * by sending the query params to the ENCODE search API.
+    *
+    * @param encodeEntity the type of ENCODE entity the stage should query
+    */
+  class EncodeLookup(encodeEntity: EncodeEntity)
+      extends ScalaAsyncLookupDoFn[List[(String, String)], String, OkHttpClient] {
+
+    private val baseParams =
+      List("frame=object", "status=released", "limit=all", "format=json")
+
+    override def asyncLookup(
+      client: OkHttpClient,
+      params: List[(String, String)]
+    ): Future[String] = {
+      val paramStrings = params.map {
+        case (key, value) =>
+          s"$key=$value"
+      }
+      val allParams = s"type=${encodeEntity.encodeApiName}" :: baseParams ::: paramStrings
```

Comment:
If `EncodeEntity` stays but all methods are deleted, this would be `encodeEntity.entryName`